static/app/components/replays/utils.tsx

```diff
@@ -28,3 +32,79 @@ export function formatTime(ms: number): string {
   }
   return `${minute}:${padZero(second)}`;
 }
+
+/**
+ * Figure out how many ticks to show in an area.
+ * If there is more space available, we can show more granular ticks, but if
+ * less space is available, fewer ticks.
+ * Similarly if the duration is short, the ticks will represent a short amount
+ * of time (like every second) but if the duration is long one tick may
+ * represent an hour.
+ *
+ * @param duration The amount of time that we need to chop up into even sections
+ * @param width Total width available, pixels
+ * @param minWidth Minimum space for each column, pixels. Ex: So we can show formatted time like `1:00:00` between major ticks
+ * @returns
+ */
+export function countColumns(duration: number, width: number, minWidth: number = 50) {
+  const maxCols = Math.ceil(width / minWidth);
+
+  // List of all the possible time granularities to display
+  // We could generate the list, which is basically a version of fizzbuzz, hard-coding is quicker.
+  const timeOptions = [
+    1 * HOUR,
+    30 * MINUTE,
+    20 * MINUTE,
+    15 * MINUTE,
+    10 * MINUTE,
+    5 * MINUTE,
+    2 * MINUTE,
+    1 * MINUTE,
+    30 * SECOND,
+    10 * SECOND,
+    5 * SECOND,
+    1 * SECOND,
+  ];
+
+  const timeBasedCols = timeOptions.reduce<Map<number, number>>((map, time) => {
+    map.set(time, Math.floor(duration / time));
+    return map;
+  }, new Map());
+
+  const [timespan, cols] = Array.from(timeBasedCols.entries())
+    .filter(([_span, c]) => c <= maxCols) // Filter for any valid timespan option where all ticks would fit
+    .reduce((best, next) => (next[1] > best[1] ? next : best), [0, 0]); // select the timespan option with the most ticks
+
+  const remaining = (duration - timespan * cols) / timespan;
+  return {timespan, cols, remaining};
+}
+
+/**
+ * Group Crumbs for display along the timeline.
+ *
+ * The timeline is broken down into columns (aka buckets, or time-slices).
+ * Columns translate to a fixed width on the screen, to prevent side-scrolling.
+ *
+ * This function groups crumbs into columns based on the number of columns available
+ * and the timestamp of the crumb.
+ */
```

Comment:
Wanted to point out this function, because it works really well thanks to css grid. 

The result of the render is this:
<img width="271" alt="Screen Shot 2022-04-25 at 11 12 33 AM" src="https://user-images.githubusercontent.com/187460/165149308-fd3e091e-9770-4d9f-84d9-e87c8cf67325.png">

You can see the purple and red lines stacked ontop of each other near the right side. That is an example of two events happening within the same time slice.